pandas/util/mixins.py

```diff
@@ -0,0 +1,28 @@
+"""
+Mixins which can't go to core.common or core.base due to circular imports
+"""
```

Comment:
AFAIR I tried importing `StringMethods` directly above the `str` declaration and it still didn't work. The utils place was really the last ressort...

Re `IndexOpsMixin` so instead of creating `utils\mixins.py` with only `NoNewAttributesMixin` you want `core\mixins.py` with only `IndexOpsMixin` (together they will create again a loop: `string.py` requires `NoNewAttributeMixin` loading that will load `IndexOpsMixin` which will require `StringMethods` in `strings.py`...